src/agent.py

```
@@ -361,9 +365,18 @@ async def generate_semantic_queries(query: str) -> list[str]:
     logger.debug("Calling LLM for query generation")
     llm_start = asyncio.get_event_loop().time()
 
+    try:
+        provider = ModelProvider(settings.LLM.QUERY_GENERATION_PROVIDER)
+    except ValueError as e:
+        logger.error(
+            f"Invalid query-generation provider '{settings.LLM.QUERY_GENERATION_PROVIDER}': {e}"
+        )
```

Ruff (0.11.9)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)